test-cypress/integration/sparql/sparql.menu.spec.js

```diff
@@ -1,807 +1,1034 @@
+            getSubmitSavedQueryBtn().click();
 
-        cy.get('.yasr_downloadIcon').should('be.visible');
+            getSavedQueryForm().should('not.exist');
 
-    });
+            // Verify that the query is saved
+            getSavedQueriesPopupBtn().click();
+            getPopover()
+                .should('be.visible')
+                .and('contain', savedQueryName);
 
-    it('Test execute query with and without "Including inferred" selected', function () {
-        let testRepo = 'test-repo' + Date.now();
-        // First navigate to create repository
-        cy.navigateToPageS('Setup', 'Repositories');
-        // // Create new repository
-        cy.createNewRepoS(testRepo).wait(1000);
-        cy.setRepoDefaultS(testRepo);
-        cy.selectRepoS(testRepo).wait(1000);
+            // Press the pen icon to edit the custom query created earlier
+            clickEditSaveQueryFromPopup(savedQueryName);
 
-        cy.navigateToPageS('SPARQL');
+            getPopover().should('not.exist');
+            getSavedQueryForm().should('be.visible');
 
-        let queryToExecute = 'INSERT DATA { <urn:a> <http://a/b> <urn:b> . <urn:b> <http://a/b> <urn:c> . }';
+            // TODO: Can make use of .within?
+            getSavedQueryNameField().should('have.value', savedQueryName);
+            getSavedQueryField()
+                .should('have.value', QUERY_FOR_SAVING)
+                .clear()
+                .type(QUERY_FOR_SAVING_MODIFIED)
+                .should('have.value', QUERY_FOR_SAVING_MODIFIED);
+            getSubmitSavedQueryBtn().click();
 
-        // Type a query with inference
-        cy.executeCustomQuery(queryToExecute);
-        // Execute default query with ‘Include inferred’ which is selected by default
-        cy.executeCustomQuery(defaultQuery);
+            getSavedQueryForm().should('not.exist');
 
-        // Confirm that all statements are available (70 from ruleset, 2 explicit and 2 inferred)
-        cy.verifyQueryResultsS(74);
+            // Verify that the query is edited.
+            // Select the query from the saved queries again
+            getSavedQueriesPopupBtn().click();
+            clickEditSaveQueryFromPopup(savedQueryName);
 
-        // Uncheck ‘Include inferred’ using the same query
-        cy.get('#inference').as('uncheckIncludeInferred');
+            getSavedQueryNameField().should('have.value', savedQueryName);
+            getSavedQueryField().should('have.value', QUERY_FOR_SAVING_MODIFIED);
 
-        cy.get('@uncheckIncludeInferred').click();
+            // Close edit saved query dialog
+            getSubmitSavedQueryBtn().click();
 
-        // Click on the Run button
-        cy.get('div#runButton > #wb-sparql-runQuery').as('rerunQuery');
+            // Click on the saved queries icon
+            getSavedQueriesPopupBtn().click();
+            // Press the trash bin to delete the custom query created earlier
+            clickDeleteSaveQueryFromPopup(savedQueryName);
```

Comment:
`deleteSaveQueryFromPopup` ? We have to discuss about the style and follow it in order to have consistency.